### PR TITLE
Add a data providers CRUD page to basic crud app example

### PR DESCRIPTION
**Pull request by prakhargupta1**
- [x] After adding a record, the Add Record button goes into a disabled state and Edit, delete buttons disappear.

https://github.com/mui/mui-toolpad/assets/92228082/0b7e079c-2c20-4123-bb55-7bc3f39fa33e

### CODE REVIEW

**Comment by prakhargupta1** [comment]
After adding a record, the Add Record button goes into a disabled state.

https://github.com/mui/mui-toolpad/assets/92228082/bc3c581c-9ea4-4866-96dc-c7b7393f99bf

File: examples/basic-crud-app/toolpad/resources/dataProvider.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  },
+
+  async createRecord(data: Customer) {
+    const highestId = records.reduce((maxId, record) => Math.max(maxId, record.id), 0);
+    // Assign a new unique id to the new record
+    const newRecord = { ...data, id: highestId + 1 };
+    return records.push(newRecord);
+  },
+
+  // async updateRecord(id, data) {},
```
Comment (Janpot) [inline examples/basic-crud-app/toolpad/resources/dataProvider.ts:103]:
```suggestion
  // async updateRecord(id, data) {},
```
?

**Review by Janpot — APPROVED** [review]
One part of commented out code that can be removed. Looks good otherwise

**Comment by prakhargupta1** [comment]
Does https://github.com/mui/mui-toolpad/pull/3215 fix the issue in the video above?

**Comment by Janpot** [comment]
> Does https://github.com/mui/mui-toolpad/pull/3215 fix the issue in the video above?

No, I'm looking into it

**Comment by prakhargupta1** [comment]
@Janpot In v0.1.50, the experience is the same as in the video above. Should I still merge this one for now?

**Comment by Janpot** [comment]
@prakhargupta1 I found the problem, you were not returning a new record from `createRecord` and the grid didn't give good feedback. I'm adding a few tweaks to the grid to improve this  behavior

**Comment by prakhargupta1** [comment]
> @prakhargupta1 I found the problem, you were not returning a new record from `createRecord` and the grid didn't give good feedback. I'm adding a few tweaks to the grid to improve this behavior

Ah! Thanks, Looks perfect now.

**Comment by Janpot** [comment]
@prakhargupta1 fixed

**Pull request merged**